elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -0,0 +1,244 @@
+                : new ArrayList<>();
+        String requestedFields = environment.field.getName() + (children.size() > 0
+                ? "(" + children.stream().map(Field::getName).collect(Collectors.toList()) + ")" : "");
+        GraphQLType parent = environment.parentType;
+        log.debug("{} {} fields for {} with parent {}<{}>",
+                operation, requestedFields, environment.source, parent.getClass().getSimpleName(), parent.getName());
+    }
+
+    private Object deleteObject(Environment request) {
+        Set<Object> deleted = new HashSet<>();
+
+        if (!request.id.isPresent() && request.data.size() > 1) {
+            throw new WebApplicationException("Id argument specification with an additional list of id's to delete "
+                    + "is unsupported", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        EntityDictionary dictionary = request.requestScope.getDictionary();
+
+        String loadType = request.field.getName();
+        // TODO: This works at the root-level, but will it blow up in nested deletes?
+        String idFieldName = dictionary.getIdFieldName(dictionary.getEntityClass(loadType));
+        String loadId = request.field.getArguments().stream()
+                .filter(arg -> ARGUMENT_DATA.equals(arg.getName()))
+                .findFirst()
+                .map(Argument::getChildren)
+                // TODO: Iterate over children and determine which contains id.
+                .map(List::toString) // TODO: place holder. remove this line.
+//                    .map(arg -> {
+//                        String specifiedId = arg.getValue().toString();
+//                        if (id != null && !id.isEmpty() && !id.equals(specifiedId)) {
+//                            throw new WebApplicationException("Specified non-matching id's as argument and data.",
+//                                    HttpStatus.SC_BAD_REQUEST);
+//                        }
+//                        return specifiedId;
+//                    })
+                .orElseGet(request.id::get);
+
+        if (loadId == null) {
+            throw new WebApplicationException("Did not specify id of object type to delete.",
+                    HttpStatus.SC_BAD_REQUEST);
+        }
+
+        PersistentResource deleteObject = load(loadType, loadId, request.requestScope);
+
+        if (deleteObject == null || deleteObject.getObject() == null) {
+            throw new WebApplicationException("Attempted to delete non-existent id.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        deleteObject.deleteResource();
+        deleted.add(deleteObject);
+
+        return deleted;
+    }
+
+    private Object createObject(Environment request) {
+        EntityDictionary dictionary = request.requestScope.getDictionary();
+
+        GraphQLObjectType objectType;
+        String uuid = UUID.randomUUID().toString();
+        if (request.outputType instanceof GraphQLObjectType) {
+            // No parent
```

Comment:
Actually we don't. The way the datafetcher works we will get a call to `get(DataFetchingEnvironment)` for each attribute that is passed in